internal/internal_workers_test.go

```diff
@@ -101,12 +121,67 @@ func (s *WorkersTestSuite) TestActivityWorker() {
 	hostEnv := getHostEnvironment()
 	hostEnv.addActivity(a.ActivityType().Name, a)
 	activityWorker := newActivityWorker(
-		s.service, domain, executionParameters, overrides, hostEnv,
+		s.service, domain, executionParameters, overrides, hostEnv, nil,
 	)
 	activityWorker.Start()
 	activityWorker.Stop()
 }
 
+func (s *WorkersTestSuite) TestActivityWorkerStop() {
+	domain := "testDomain"
+	logger, _ := zap.NewDevelopment()
+
+	pats := &m.PollForActivityTaskResponse{
+		TaskToken: []byte("token"),
+		WorkflowExecution: &m.WorkflowExecution{
+			WorkflowId: common.StringPtr("wID"),
+			RunId:      common.StringPtr("rID")},
+		ActivityType:                  &m.ActivityType{Name: common.StringPtr("test")},
+		ActivityId:                    common.StringPtr(uuid.New()),
+		ScheduledTimestamp:            common.Int64Ptr(time.Now().UnixNano()),
+		ScheduleToCloseTimeoutSeconds: common.Int32Ptr(1),
+		StartedTimestamp:              common.Int64Ptr(time.Now().UnixNano()),
+		StartToCloseTimeoutSeconds:    common.Int32Ptr(1),
+		WorkflowType: &m.WorkflowType{
+			Name: common.StringPtr("wType"),
+		},
+		WorkflowDomain: common.StringPtr("domain"),
+	}
+
+	s.service.EXPECT().DescribeDomain(gomock.Any(), gomock.Any(), callOptions...).Return(nil, nil)
+	s.service.EXPECT().PollForActivityTask(gomock.Any(), gomock.Any(), callOptions...).Return(pats, nil).AnyTimes()
+	s.service.EXPECT().RespondActivityTaskCompleted(gomock.Any(), gomock.Any(), callOptions...).Return(nil).AnyTimes()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	stopChannel := make(chan struct{})
+	executionParameters := workerExecutionParameters{
+		TaskList:                        "testTaskList",
+		ConcurrentPollRoutineSize:       5,
+		ConcurrentActivityExecutionSize: 2,
+		Logger:                          logger,
+		UserContext:                     ctx,
+		UserContextCancel:               cancel,
+		WorkerStopTimeout:               time.Second * 2,
+	}
+	overrides := &workerOverrides{activityTaskHandler: newNoResponseActivityTaskHandler()}
+	a := &greeterActivity{}
+	hostEnv := getHostEnvironment()
+	hostEnv.addActivity(a.ActivityType().Name, a)
+	activityWorker := newActivityWorker(
+		s.service, domain, executionParameters, overrides, hostEnv, stopChannel,
+	)
+	activityWorker.Start()
+	time.Sleep(1 * time.Second)
```

Comment:
Could you refactor the test to not rely on Sleep? It adds 2 seconds to the overall test suite execution.

Reply:
Good point. Updated this test to get rid of sleep.